migrations/1666703991492_pox_events.js

```diff
@@ -0,0 +1,190 @@
+    name: {
+      type: 'string',
+      notNull: true,
+    },
+    pox_addr: {
+      type: 'string',
+    },
+    pox_addr_raw: {
+      type: 'bytea',
+    },
+    first_cycle_locked: { // unique to handle-unlock
+      type: 'numeric',
+    },
+    first_unlocked_cycle: { // unique to handle-unlock
+      type: 'numeric',
+    },
+    lock_period: { // unique to stack-stx, delegate-stack-stx
+      type: 'numeric'
+    },
+    lock_amount: { // unique to stack-stx, delegate-stack-stx
+      type: 'numeric'
+    },
+    start_burn_height: { // unique to stack-stx, delegate-stack-stx
+      type: 'numeric',
+    },
+    unlock_burn_height: { // unique to stack-stx, stack-extend, delegate-stack-stx, delegate-stack-extend
+      type: 'numeric',
+    },
+    delegator: { // unique to delegate-stack-stx, delegate-stack-increase, delegate-stack-extend
+      type: 'string',
+    },
+    increase_by: { // unique to stack-increase, delegate-stack-increase
+      type: 'numeric',
+    },
+    total_locked: { // unique to stack-increase, delegate-stack-increase
+      type: 'numeric',
+    },
+    extend_count: { // unique to stack-extend, delegate-stack-extend
+      type: 'numeric',
+    },
+    reward_cycle: { // unique to stack-aggregation-*
+      type: 'numeric',
+    },
+    amount_ustx: { // unique to stack-aggregation-*
+      type: 'numeric',
+    },
+  });
+
+  pgm.addConstraint('pox2_events', 'valid_event_specific_columns', `CHECK (
+    CASE name 
+      WHEN 'handle-unlock' THEN 
+        first_cycle_locked IS NOT NULL AND 
+        first_unlocked_cycle IS NOT NULL
+      WHEN 'stack-stx' THEN
+        lock_period IS NOT NULL AND 
+        lock_amount IS NOT NULL AND 
+        start_burn_height IS NOT NULL AND 
+        unlock_burn_height IS NOT NULL
+      WHEN 'stack-increase' THEN
+        increase_by IS NOT NULL AND 
+        total_locked IS NOT NULL
+      WHEN 'stack-extend' THEN
+        extend_count IS NOT NULL AND 
+        unlock_burn_height IS NOT NULL
+      WHEN 'delegate-stack-stx' THEN
+        lock_period IS NOT NULL AND 
+        lock_amount IS NOT NULL AND 
+        start_burn_height IS NOT NULL AND 
+        unlock_burn_height IS NOT NULL AND 
+        delegator IS NOT NULL
+      WHEN 'delegate-stack-increase' THEN
+        increase_by IS NOT NULL AND 
+        total_locked IS NOT NULL AND
+        delegator IS NOT NULL
+      WHEN 'delegate-stack-extend' THEN
+        extend_count IS NOT NULL AND 
+        unlock_burn_height IS NOT NULL AND 
+        delegator IS NOT NULL
+      WHEN 'stack-aggregation-commit' THEN
+        reward_cycle IS NOT NULL AND 
+        amount_ustx IS NOT NULL
+      WHEN 'stack-aggregation-commit-indexed' THEN
+        reward_cycle IS NOT NULL AND 
+        amount_ustx IS NOT NULL
+      WHEN 'stack-aggregation-increase' THEN
+        reward_cycle IS NOT NULL AND 
+        amount_ustx IS NOT NULL
+      ELSE false
+    END
+  )`);
+
+  pgm.createIndex('pox2_events', 'block_height');
```

Comment:
This would be much more performant as 
```js
pgm.createIndex('txs', [
  { name: 'block_height', sort: 'DESC' },
  { name: 'microblock_sequence', sort: 'DESC' },
  { name: 'tx_index', sort: 'DESC' },
  { name: 'event_index', sort: 'DESC' },
]);
  ```

Reply:
Good call, done!